inbox/mailsync/backends/imap/generic.py

```diff
@@ -118,8 +119,10 @@ def __init__(self, account_id, namespace_id, folder_name,
 
         with session_scope(namespace_id) as db_session:
             try:
+                folder_stored_name = folder_name\
+                    .rstrip()[:MAX_FOLDER_NAME_LENGTH]
                 folder = db_session.query(Folder). \
-                    filter(Folder.name == folder_name,
+                    filter(Folder.name == folder_stored_name,
```

Comment:
Oh nice catch—I wonder if there's a way we could perform this mapping lower down in the Folder model so we don't have to remember to do it all the time? Would be nice if we could say "limit the length of the comparator in the where clause to the max length of the field"...